File: 10812.py
N, M = map(int, input().split())
baskets = [i for i in range(N + 1)]

for i in range(M):
    I, J, K = map(int, input().split())

    tmp = [baskets[j] for j in range(I, K)]

    dis = K - I;
    for j in range(K, J + 1):
        baskets[j - dis] = baskets[j]
        index = j
    index += 1

    tmp_index = 0; index -= dis
    for j in range(index, J + 1):
        baskets[j] = tmp[tmp_index]
        tmp_index += 1

for basket in baskets[1:]:
    print(basket, end=' ')
print()

File: 10812.cpp
#include <iostream>
#include <vector>
using namespace std;

int baskets[101];

int main() {
	int N, M; cin >> N >> M;
	for (int i = 1; i <= N; i++)
		baskets[i] = i;

	for (int i = 0; i < M; i++) {
		int I, J, K; cin >> I >> J >> K;

		vector<int> tmp;
		for (int j = I; j < K; j++)
			tmp.push_back(baskets[j]);

		int dis = K - I; int index;
		for (index = K; index <= J; index++)
			baskets[index - dis] = baskets[index];

		int tmp_index = 0; index -= dis;
		for (; index <= J; index++, tmp_index++)
			baskets[index] = tmp[tmp_index];
	}

	for (int i = 1; i <= N; i++)
		cout << baskets[i] << ' ';
	cout << endl;
}
